PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/util/BeforeRunTaskUtils.java

```diff
@@ -0,0 +1,200 @@
+                                                                   @NotNull MavenProject mavenProject,
+                                                                   RunConfiguration runConfiguration,
+                                                                   boolean add) throws IllegalAccessException {
+        String pomXmlPath = MavenUtils.getMavenModulePath(mavenProject);
+        updateBeforeRunOption(runConfigurationEditorComponent, MavenBeforeRunTask.class, task ->
+                                      Objects.nonNull(task) && StringUtils.equals(task.getProjectPath(), pomXmlPath)
+                                              && StringUtils.equals(MAVEN_TASK_PACKAGE, task.getGoal()),
+            () -> {
+                MavenBeforeRunTask task = new MavenBeforeRunTask();
+                task.setEnabled(true);
+                task.setProjectPath(pomXmlPath);
+                task.setGoal(MAVEN_TASK_PACKAGE);
+                return task;
+            }, runConfiguration, add);
+    }
+
+    public static void addOrRemoveBuildGradleProjectBeforeRunOption(@NotNull JComponent runConfigurationEditorComponent,
+                                                                    @NotNull ExternalProjectPojo gradleProject,
+                                                                    RunConfiguration runConfiguration, boolean add)
+            throws IllegalAccessException {
+        updateBeforeRunOption(runConfigurationEditorComponent, ExternalSystemBeforeRunTask.class, task ->
+                                      Objects.nonNull(task) &&
+                                              Objects.nonNull(task.getTaskExecutionSettings())
+                                              && StringUtils.equals(task.getTaskExecutionSettings()
+                                                                        .getExternalProjectPath(),
+                                                                    gradleProject.getPath())
+                                              && CollectionUtils.isEqualCollection(task.getTaskExecutionSettings().getTaskNames(),
+                                                                                   Arrays.asList(GRADLE_TASK_ASSEMBLE)),
+            () -> {
+                GradleBeforeRunTaskProvider provider = new GradleBeforeRunTaskProvider(
+                        runConfiguration.getProject());
+                ExternalSystemBeforeRunTask task = provider.createTask(runConfiguration);
+                task.getTaskExecutionSettings()
+                    .setExternalSystemIdString(GradleConstants.SYSTEM_ID.toString());
+                task.getTaskExecutionSettings().setExternalProjectPath(gradleProject.getPath());
+                task.getTaskExecutionSettings().setTaskNames(Arrays.asList(GRADLE_TASK_ASSEMBLE));
+                return task;
+            }, runConfiguration, add);
+    }
+
+    public static <T extends BeforeRunTask> void updateBeforeRunOption(
```

Comment:
change to `addOrRemoveBuildGradleProjectBeforeRunOption` to keep consistent with previous methods?